aten/src/ATen/native/xnnpack/Engine.h

```diff
@@ -6,6 +6,90 @@ namespace at {
 namespace native {
 namespace xnnpack {
 
+//
+// Clamp
+//
+
+bool use_clamp(
+    const Tensor& input,
+    float output_min,
+    float output_max);
+
+Tensor clamp(
+    const Tensor& input,
+    float output_min,
+    float output_max);
+
+bool use_clamp_(
+    Tensor& input,
+    float output_min,
+    float output_max);
+
+Tensor& clamp_(
+    Tensor& input,
+    float output_min,
+    float output_max);
+
+bool use_clamp_out(
+    Tensor& output,
+    const Tensor& input,
+    float output_min,
+    float output_max);
+
+Tensor& clamp_out(
+    Tensor& output,
+    const Tensor& input,
+    float output_min,
+    float output_max);
+
```

Comment:
The following two functions are a handful, but I am introducing complexity here for the benefit of minimal modifications to the call sites.